File: problem_1.py
import re

def validate_username(username):
    if not username:
        return False, "Username is invalid."
    if len(username) > 50:
        return False, "Username is invalid."
    return True, "Username is valid."

def validate_password(password):
    if len(password) < 8:
        return False, "Password is invalid."
    if not re.search(r'[!@#$%^&*(),.?":{}|<>]', password):
        return False, "Password is invalid."
    if not re.search(r'\d', password):
        return False, "Password is invalid."
    if not re.search(r'[A-Z]', password):
        return False, "Password is invalid."
    if not re.search(r'[a-z]', password):
        return False, "Password is invalid."
    return True, "Password is valid."

def validate_email(email):
    pattern = r'^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
    if not re.match(pattern, email):
        return False, "Email is invalid."
    return True, "Email is valid."

def main():
    username = input("Enter Username: ")
    password = input("Enter Password: ")
    email = input("Enter Email: ")

    username_valid, username_message = validate_username(username)
    print(username_message)

    password_valid, password_message = validate_password(password)
    print(password_message)

    email_valid, email_message = validate_email(email)
    print(email_message)



if __name__ == "__main__":
    main()

File: Problem1.java
// Problem 1: User Login Validation

import java.util.Scanner;
import java.util.regex.Pattern;

public class Problem1 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Prompt for username input
        System.out.print("Enter username: ");
        String username = scanner.nextLine();

        // Prompt for password input
        System.out.print("Enter password: ");
        String password = scanner.nextLine();

        // Prompt for email input
        System.out.print("Enter email: ");
        String email = scanner.nextLine();

        // Validate username
        if (validateUsername(username)) {
            System.out.println("Username is valid.");
        } else {
            System.out.println("Username is invalid.");
        }

        // Validate password
        if (validatePassword(password)) {
            System.out.println("Password is valid.");
        } else {
            System.out.println("Password is invalid.");
        }

        // Validate email
        if (validateEmail(email)) {
            System.out.println("Email is valid.");
        } else {
            System.out.println("Email is invalid.");
        }
        
        scanner.close();
    }

    // Validate username method
    public static boolean validateUsername(String username) {
        if (username.isEmpty()) {
            return false;
        }
        if (username.length() > 50) {
            return false;
        }
        return true;
    }

    // Validate password method
    public static boolean validatePassword(String password) {
        if (password.length() < 8) {
            return false;
        }
        if (!Pattern.compile("[!@#$%^&*(),.?\":{}|<>]").matcher(password).find()) {
            return false;
        }
        if (!Pattern.compile("\\d").matcher(password).find()) {
            return false;
        }
        if (!Pattern.compile("[A-Z]").matcher(password).find()) {
            return false;
        }
        if (!Pattern.compile("[a-z]").matcher(password).find()) {
            return false;
        }
        return true;
    }

    // Validate email method 
    public static boolean validateEmail(String email) {
        if (!Pattern.compile("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$").matcher(email).matches()) {
          
            return false;
        }
        return true;
    }
}
